src/Types.jl

```diff
@@ -207,15 +207,15 @@ function projectfile_path(env_path::String; strict=false)
         maybe_file = joinpath(env_path, name)
         isfile(maybe_file) && return maybe_file
     end
-    return strict ? nothing : joinpath(env_path, "Project.toml")
+    return strict ? nothing : joinpath(env_path, Base.project_names[end])
```

Comment:
Good point, I'll change that. When editing that function I somehow found `project_files[end]` was used in a very similar way in that file:
```
$ git grep -n "project_names\[end\]" -- src/
src/Types.jl:232:            project_file = joinpath(env, Base.project_names[end])
src/Types.jl:235:                abspath(env, Base.project_names[end])
```

Reply:
What about the other occurrences?
```
$ git grep -n "Project\.toml" -- src/
src/Artifacts.jl:755:        # Does a `Project.toml` file exist here, in the absence of an Artifacts.toml?
src/Artifacts.jl:758:        if isfile(joinpath(path, "Project.toml"))
src/Pkg.jl:281:Otherwise, resolve a set of feasible packages from the `Project.toml` files
src/Pkg.jl:455:    touch("Project.toml")
src/Types.jl:210:    return strict ? nothing : joinpath(env_path, "Project.toml")
src/backwards_compatible_isolation.jl:517:        localctx.env.project_file = joinpath(tmpdir, "Project.toml")
src/generate.jl:53:    genfile(pkg, dir, "Project.toml"; preview=preview) do io
```

Reply:
For the sake of consistency, you can change all of them to `Base.project_names[end]` :+1:.

Reply:
We can fix that in another PR.

Reply:
:thinking: I'm pretty sure the `Artifacts` occurrence is a bug. I think it should be `isfile(joinpath(path, "Project.toml"))` -> `projectfile_path(path;strict=true) !== nothing`

Reply:
1. RE `src/Artifacts.jl`: I can't find any documentation of an `Artifacts.toml` file in [Julia's docs](https://docs.julialang.org/en/v1/search/?q=artifacts) and in [Pkg's docs](https://julialang.github.io/Pkg.jl/v1/search/?q=artifacts), so I would leave it untouched as there might be a bigger change needed.
2. RE `src/Pkg.jl`: The first one is a doc storing (I could change that to something like `$(Base.project_names[end])`), the second one I don't know.
3. RE `src/Types.jl:210`: ... as already mentioned
4. RE `src/backwards_compatible_isolation.jl`: same type of usage for both Project and Manifest files, so they should be changed, too?
5. RE `src/generate.jl`: same here

And while thinking about that, should I also change the occurrences of `"Manifest.toml"`?

```
$ git grep -n "Manifest\.toml" -- src/
src/API.jl:272:`Manifest.toml` and `Artifacts.toml` files, noting those that have been deleted, and then
src/API.jl:283:    # "index files" (`Manifest.toml`, `Artifacts.toml`) were last used.  We will combine
src/API.jl:363:    all_manifest_files = filter(f -> endswith(f, "Manifest.toml"), all_index_files)
src/API.jl:364:    all_artifacts_files = filter(f -> !endswith(f, "Manifest.toml"), all_index_files)
src/Pkg.jl:279:If a `Manifest.toml` file exists in the current project, download all
src/Types.jl:218:    return strict ? nothing : replace(projectfile_path(env_path), r"Project\.toml$" => "Manifest.toml")
src/backwards_compatible_isolation.jl:518:        localctx.env.manifest_file = joinpath(tmpdir, "Manifest.toml")
```
If you think that worthwhile then I'll open another PR. The scope is different and this one has already been approved.